packages/structures/src/users/User.ts

```diff
@@ -0,0 +1,178 @@
+		super(data);
+	}
+
+	/**
+	 * {@inheritDoc Structure._patch}
+	 *
+	 * @internal
+	 */
+	public override _patch(data: Partial<APIUser>) {
+		return super._patch(data);
+	}
+
+	/**
+	 * The user's id
+	 */
+	public get id() {
+		return this[kData].id;
+	}
+
+	/**
+	 * The username of the user
+	 */
+	public get username() {
+		return this[kData].username;
+	}
+
+	/**
+	 * The user's 4 digit tag, if a bot or not migrated to unique usernames
+	 */
+	public get discriminator() {
+		return this[kData].discriminator;
+	}
+
+	/**
+	 * The user's display name, the application name for bots
+	 */
+	public get globalName() {
+		return this[kData].global_name;
+	}
+
+	/**
+	 * The name displayed in the client for this user when no nickname is set
+	 */
+	public get displayName() {
+		return this.globalName ?? this.username;
+	}
+
+	/**
+	 * The user avatar's hash
+	 */
+	public get avatar() {
+		return this[kData].avatar;
+	}
+
+	/**
+	 * Whether the user is a bot
+	 */
+	public get bot() {
+		return this[kData].bot ?? false;
+	}
+
+	/**
+	 * Whether the user is an Official Discord System user
+	 */
+	public get system() {
+		return this[kData].system ?? false;
+	}
+
+	/**
+	 * Whether the user has mfa enabled
+	 *
+	 * @remarks This property is only set when the user was fetched with an OAuth2 token and the `identify` scope
+	 */
+	public get mfaEnabled() {
+		return this[kData].mfa_enabled;
+	}
+
+	/**
+	 * The user's banner hash
+	 *
+	 * @remarks This property is only set when the user was manually fetched
+	 */
+	public get banner() {
+		return this[kData].banner;
+	}
+
+	/**
+	 * The base 10 accent color of the user's banner
+	 *
+	 * @remarks This property is only set when the user was manually fetched
+	 */
+	public get accentColor() {
+		return this[kData].accent_color;
+	}
+
+	/**
+	 * The user's primary discord language
+	 *
+	 * @remarks This property is only set when the user was fetched with an Oauth2 token and the `identify` scope
+	 */
+	public get locale() {
+		return this[kData].locale;
+	}
+
+	/**
+	 * Whether the email on the user's account has been verified
+	 *
+	 * @remarks This property is only set when the user was fetched with an OAuth2 token and the `email` scope
+	 */
+	public get verified() {
+		return this[kData].verified;
+	}
+
+	/**
+	 * The user's email
+	 *
+	 * @remarks This property is only set when the user was fetched with an OAuth2 token and the `email` scope
+	 */
+	public get email() {
+		return this[kData].email;
+	}
+
+	/**
+	 * The type of nitro subscription on the user's account
+	 *
+	 * @remarks This property is only set when the user was fetched with an OAuth2 token and the `identify` scope
+	 */
+	public get premiumType() {
+		return this[kData].premium_type;
+	}
+
+	/**
+	 * The timestamp the user was created at
+	 */
+	public get createdTimestamp() {
+		return typeof this.id === 'string' || typeof this.id === 'bigint' ? DiscordSnowflake.timestampFrom(this.id) : null;
```

Comment:
If this becomes a common check, please make a typeguard function for this and reuse it

```ts
function isIdSet(id: unknown): id is string | bigint {
	return typeof id === 'string' || typeof id === 'bigint';
}
```

Reply:
as standalone function in utils/ you mean?